### PR TITLE
Fix ability to set on_demand to false and on_hand

### PR DESCRIPTION
**Pull request by frank-west-iii**
When setting `on_demand` to false and also entering a value for
`on_hand` the guard was restricting the setting of the value because
`on_demand` was not yet changed.

Since we guard against the showing of the `on_hand` value by checking
the `on_demand` value there is no harm in allowing this value to be
updated. Currently the previous `unit_on_hand` value is still set in the
database and that is what gets shown to the user after an update occurs.

Fixes https://github.com/openfoodfoundation/openfoodnetwork/issues/2351

### CODE REVIEW

**Comment by frank-west-iii** [comment]
@sauloperez, looks like I didn't base this branch off the same one you did for your 
PRs. I chose the branch in the main repo's `Gemfile`. Not sure if specs should be working or not.

**Review by mkllnk — COMMENTED** [review]
The change looks good.

I think we are working on branch 2-0-4-stable at the moment and unfortunately tests are not passing yet.

**Comment by myriamboure** [comment]
This has been standing for a while, and hard to follow as not in the main repo. Can other devs give a review so that we can move it forward? How to fix the test issue? @mkllnk @sauloperez ?

**Comment by mkllnk** [comment]
We are working on fixing the test issue for the Spree upgrade. We should probably apply this to the Spree upgrade, but I would like to see Pau's opinion on that.

**Comment by sauloperez** [comment]
Sorry, I forgot about this one. It's hard for me to follow all GH notifications.

I would merge this into this exact `step-6a` branch so that we can get the fix without having to wait for the spree upgrade.

I tried to run `spec/models/spree/variant_spec.rb` in my machine but I encountered too many yak shavings to set up the testing dummy app. Could you try yourself @frank-west-iii or someone else? We need to make sure we're breaking any tests.

**Comment by frank-west-iii** [comment]
I'm heading out on holiday until August 1st. If it still needs to be done when I get back I can do it.

**Review by mkllnk — APPROVED** [review]
I created a branch of openfoodnetwork to test this commit and that passes: https://semaphoreci.com/mkllnk/openfoodnetwork/branches/update-spree-patch/builds/2.

I think we can merge it and create a pull request on openfoodnetwork to use the new version.

**Comment by sauloperez** [comment]
Did you manage to see what's failing from Spree's test suite? I'm not very confident seeing it fail...

**Comment by mkllnk** [comment]
No, I didn't check that. And I'm not sure if I have time to get the test suite running.

**Comment by sauloperez** [comment]
Right, we never had CI for our Spree fork before enabling it for the upgrade, so nevermind.

**Comment by myriamboure** [comment]
@mkllnk @sauloperez so we merged but it has not been tested right? How do we know it is solving our original issue https://github.com/openfoodfoundation/openfoodnetwork/issues/2351?

**Comment by sauloperez** [comment]
it required some manual work to get it working but we should test again now that is merged @myriamboure. We might have cheated the process here a bit, yes.

**Comment by mkllnk** [comment]
@myriamboure a little explanation how Spree and OFN pull requests are connected. The OFN always depends on a specific version of Spree. Let's say it's version `a`. This pull request changed Spree and made it version `b`. This doesn't change the OFN as it's still using `a`. Pau opened https://github.com/openfoodfoundation/openfoodnetwork/pull/2509 to make OFN use the new version `b`. That version of the OFN needs to be tested. We can't test a Spree change just on it's own.
